lib/initialize.js

```diff
@@ -0,0 +1,86 @@
+const path = require('path')
+const os = require("os")
+const rimraf = require('rimraf')
+const fs = require("fs")
+
+const {createCustomFileAccessor} = require("./config")
+
+const appDataPath = path.join(os.homedir(), 'AppData', 'Roaming', 'parsec-switcher');
+
+const configPath = path.join(appDataPath,'config.json')
+
+const dataPath = path.join(appDataPath,'account_data.json')
+
+const multiUserPath = path.join(appDataPath,'MultiUser')
+
+const defaultUserPath = path.join(multiUserPath,'default')
+
+const parsecUser = path.join(appDataPath,'..','Parsec','user.bin')
+
+
+const onConfigChanged = []
+
+const global_state = {}
+
+
+function initialize(){
+    const initialFolders = [appDataPath,multiUserPath]
+    const runOnConfigChanged = ()=>onConfigChanged.forEach((run) => run())
+    const config = createCustomFileAccessor(configPath,runOnConfigChanged)
+    const data = createCustomFileAccessor(dataPath,runOnConfigChanged)
+
+    let makeDefault = 0
+
+    if(!config["setupNotRequired"]){
+        //todo logger
+        console.log("inside first time setup")
+
+        if(fs.existsSync(parsecUser)){
+            makeDefault  = 1
+            initialFolders.push(defaultUserPath)
+        }
+
+        initialFolders.forEach((folder) =>{
+            rimraf.sync(folder)
+            fs.mkdirSync(folder)
+        })
+
+        if(makeDefault){
+            console.log("Making default user")
+            data["default"] = path.join(defaultUserPath,'user.bin')
+            config["currentUser"] = "default"
+        }
+
+        const parsecdDefaultLocation = "C:\\Program Files\\Parsec\\parsecd.exe"
+        if(fs.existsSync(parsecdDefaultLocation)){
+            config["parsecdLocation"] = parsecdDefaultLocation
+            config["parsecdFound"] = 1
+        }
+        else {
+            config["parsecdFound"] = 0
```

Comment:
I dont see the need of maintaining `parsecdFound` in config, as we can check it exists during runtime... which will also cover the case if incase someone moves the parsec exe

Reply:
moved this to flags object